js/ui/data_grid.d.ts

```diff
@@ -523,168 +674,32 @@ export interface GridBaseOptions<T = GridBase> extends WidgetOptions<T> {
     focusedRowKey?: any;
     /**
      * @docid
+     * @type object
      * @prevFileNamespace DevExpress.ui
      * @public
      */
-    headerFilter?: {
+    headerFilter?: HeaderFilter;
     /**
-       * @docid
-       * @prevFileNamespace DevExpress.ui
-     * @default false
-       */
-      allowSearch?: boolean,
-      /**
-       * @docid
-     * @prevFileNamespace DevExpress.ui
-       * @default 315 [for](Material)
-       * @default 325
-       */
-      height?: number,
-      /**
-       * @docid
-       * @prevFileNamespace DevExpress.ui
-       * @default 500
-       */
-      searchTimeout?: number,
-      /**
-       * @docid
-       * @prevFileNamespace DevExpress.ui
-       */
-      texts?: {
-        /**
-         * @docid
-         * @prevFileNamespace DevExpress.ui
-         * @default "Cancel"
-         */
-        cancel?: string,
-        /**
-         * @docid
-         * @prevFileNamespace DevExpress.ui
-         * @default "(Blanks)"
-         */
-        emptyValue?: string,
-        /**
-         * @docid
-         * @prevFileNamespace DevExpress.ui
-         * @default "Ok"
-         */
-        ok?: string
-      },
-      /**
-       * @docid
-       * @prevFileNamespace DevExpress.ui
-       * @default false
-       */
-      visible?: boolean,
-      /**
-       * @docid
-       * @prevFileNamespace DevExpress.ui
-       * @default 252
-       */
-      width?: number
-    };
-    /**
-     * @docid
+     * @docid
      * @default false
      * @prevFileNamespace DevExpress.ui
      * @public
      */
     highlightChanges?: boolean;
     /**
      * @docid
+     * @type object
      * @prevFileNamespace DevExpress.ui
      * @public
      */
-    keyboardNavigation?: {
-    /**
-       * @docid
-     * @prevFileNamespace DevExpress.ui
-       * @default false
-       */
-      editOnKeyPress?: boolean,
-      /**
-       * @docid
-       * @prevFileNamespace DevExpress.ui
-       * @default true
-       */
-      enabled?: boolean,
-      /**
-       * @docid
-       * @prevFileNamespace DevExpress.ui
-       * @type Enums.GridEnterKeyAction
-       * @default "startEdit"
-       */
-      enterKeyAction?: 'startEdit' | 'moveFocus',
-      /**
-       * @docid
-       * @prevFileNamespace DevExpress.ui
-       * @type Enums.GridEnterKeyDirection
-       * @default "none"
-       */
-      enterKeyDirection?: 'none' | 'column' | 'row'
-    };
+    keyboardNavigation?: GridBaseKeyboardNavigation;
```

Comment:
Rename GridBaseKeyboardNavigation to KeyboardNavigation

Reply:
done